service/namespace_service.go

```diff
@@ -0,0 +1,51 @@
+package service
+
+import (
+	"context"
+
+	"github.com/odpf/optimus/models"
+	"github.com/odpf/optimus/store"
+)
+
+type NamespaceRepoFactory interface {
+	New(spec models.ProjectSpec) store.NamespaceRepository
+}
+
+type NamespaceService interface {
+	GetProjectAndNamespace(context.Context, string, string) (models.ProjectSpec, models.NamespaceSpec, error)
+}
+
+type namespaceService struct {
+	projectService   ProjectService
+	namespaceRepoFac NamespaceRepoFactory
+}
+
+func NewNamespaceService(projectService ProjectService, factory NamespaceRepoFactory) *namespaceService {
+	return &namespaceService{
+		projectService:   projectService,
+		namespaceRepoFac: factory,
+	}
+}
+
+// GetProjectAndNamespace This function is inefficient, it gets the project from repo along with secrets
+// Then we only use the id from the project to fetch the Namespace, along with project and secrets
+// Repository can provide a method to query both together.
+func (s namespaceService) GetProjectAndNamespace(ctx context.Context, projectName string, namespaceName string) (models.ProjectSpec, models.NamespaceSpec, error) {
```

Comment:
It should just return namespace spec & the callers currently can refer to project spec inside namespace spec right?

Reply:
main concern is namespace service returning projectspec  directly